deploy/helm/kubecf/assets/operations/instance_groups/app_autoscaler.yaml

```diff
@@ -0,0 +1,1150 @@
+    type: certificate
+    options:
+      ca: servicebroker_ca
+      common_name: servicebroker client
+      extended_key_usage:
+      - client_auth
+- type: replace
+  path: /variables/-
+  value:
+    name: servicebroker_public_ca
+    type: certificate
+    options:
+      common_name: servicebrokerCA
+      is_ca: true
+- type: replace
+  path: /variables/-
+  value:
+    name: servicebroker_public_server
+    type: certificate
+    options:
+      ca: servicebroker_public_ca
+      common_name: autoscalerservicebroker.((system_domain))
+      extended_key_usage:
+      - client_auth
+      - server_auth
+- type: replace
+  path: /variables/-
+  value:
+    name: scheduler_ca
+    type: certificate
+    options:
+      common_name: schedulerCA
+      is_ca: true
+- type: replace
+  path: /variables/-
+  value:
+    name: scheduler_server
+    type: certificate
+    options:
+      ca: scheduler_ca
+      common_name: autoscalerscheduler.service.cf.internal
+      extended_key_usage:
+      - client_auth
+      - server_auth
+- type: replace
+  path: /variables/-
+  value:
+    name: scheduler_client
+    type: certificate
+    options:
+      ca: scheduler_ca
+      common_name: scheduler client
+      extended_key_usage:
+      - client_auth
+- type: replace
+  path: /variables/-
+  value:
+    name: metricsserver_ca
+    type: certificate
+    options:
+      common_name: metricsserverCA
+      is_ca: true
+- type: replace
+  path: /variables/-
+  value:
+    name: metricsserver_server
+    type: certificate
+    options:
+      alternative_names:
+      - metricsserver.service.cf.internal
+      - '*.asmetrics.default.app-autoscaler.bosh'
+      ca: metricsserver_ca
+      common_name: metricsserver.service.cf.internal
+      extended_key_usage:
+      - client_auth
+      - server_auth
+- type: replace
+  path: /variables/-
+  value:
+    name: metricsserver_client
+    type: certificate
+    options:
+      ca: metricsserver_ca
+      common_name: metricsserver client
+      extended_key_usage:
+      - client_auth
+- type: replace
+  path: /variables/-
+  value:
+    name: postgres_ca
+    type: certificate
+    options:
+      common_name: postgresCA
+      is_ca: true
+- type: replace
+  path: /variables/-
+  value:
+    name: postgres_server
+    type: certificate
+    options:
+      ca: postgres_ca
+      common_name: autoscalerpostgres.service.cf.internal
+      extended_key_usage:
+      - client_auth
+      - server_auth   
+
+# Set the correct kube addresses
+- type: replace
+  path: /instance_groups/name=asnozzle/jobs/name=metricsgateway/properties/autoscaler/metricsgateway/nozzle/rlp_addr?
+  value: {{ .Values.deployment_name }}-log-api:8082
+- type: replace
+  path: /variables/name=eventgenerator_server/options?/alternative_names
+  value: ["{{ .Values.deployment_name }}-asmetrics","*.{{ .Values.deployment_name }}-asmetrics","{{ .Values.deployment_name }}-asmetrics-0","{{ .Values.deployment_name }}-asmetrics-1"]
```

Comment:
Could you use the YAML syntax here and below?

```
- type: replace
  path: /variables/name=eventgenerator_server/options?/alternative_names
  value:
  - "{{ .Values.deployment_name }}-asmetrics"
  - "*.{{ .Values.deployment_name }}-asmetrics"
  ...
```

Reply:
Having to add the service index here sounds wrong. It won't be scalable if one wants to deploy with >2 replicas.

Reply:
Another service will communicate with this service using the bosh links instance address .Like https://github.com/cloudfoundry/app-autoscaler-release/blob/37f28f86d797c74572200f5b5750b65bb75b64a8/jobs/metricsgateway/templates/metricsgateway.yml.erb#L17 .   And the cf-operator consumes the bosh links instance address like `kubecf-asmetrics-0` 
 If removed, it will report SSL certificate error. 
```
{"data":{"error":"x509: certificate is valid for metricsserver.service.cf.internal, kubecf-asmetrics, *.kubecf-asmetrics, not kubecf-asmetrics-0","metricServerURL":"wss://kubecf-asmetrics-0:7103/v1/envelopes","session":"2"},"log_level":2,"log_time":"2019-11-13T11:11:14Z","message":"metricsgateway.WSHelper.failed-to-create-websocket-connection-to-metricserver","source":"metricsgateway","timestamp":"1573643474.563703775"}
```  
To resolve this issue, I added a pre_render_script to change the service address to `<instance.address>.<namespace>.svc.cluster.local` and also add a domain name `*.{{ .Release.Namespace }}.svc.cluster.local` to the cert alternative_names. 

if you have any other suggestions here it will be very appreciated .

Reply:
For now, we can maybe control the number of replicas from `values.yaml` and add the indexed service names to the certificate.

Reply:
This one has changed as you suggested.
```
{{- $root := . -}}
{{- range $index, $val := until (int .Values.replicas ) }}
- type: replace
  path: /variables/name=metricsserver_server/options?/alternative_names?/-
  value: "{{$root.Values.deployment_name}}-asmetrics-{{$val}}"
{{- end }}
```